discord/client.py

```diff
@@ -304,8 +304,8 @@ async def __aexit__(
         exc_value: Optional[BaseException],
         traceback: Optional[TracebackType],
     ) -> None:
-        if not self.is_closed():
-            await self.close()
+        # Don't double-call user implementations
+        await (self.close() if self._closing_task is None else self._closing_task)
```

Comment:
Can you expand this out instead of using the ternary?

Reply:
sure